### PR TITLE
[Fusion] use paddle.nn.functional.swiglu to replace manual chunk+silu

### PR DESCRIPTION
**Pull request by huangjiyi**
## Summary
- Replace manual `paddle.chunk` + `F.silu` implementation in `swiglu()` with the native `paddle.nn.functional.swiglu()` API for cleaner code and potential performance benefit.

## Test plan
- [ ] Verify existing unit tests pass with the new implementation
- [ ] Confirm `paddle.nn.functional.swiglu` is available in the target Paddle version

### CODE REVIEW

**Comment by huangjiyi** [comment]
@ShigureNyako 帮我 Review 一下这个 PR

File: src/paddlefleet/fusions/fused_bias_swiglu.py [hunk 1]
```diff
@@ -38,8 +37,7 @@ def swiglu(y):
     Returns:
         paddle.Tensor: Result of SwiGLU activation: SiLU(y1) * y2, where y1, y2 are the split halves.
     """
-    y_1, y_2 = paddle.chunk(y, 2, -1)
-    return F.silu(y_1) * y_2
+    return paddle.nn.functional.swiglu(y)
```
Comment (ShigureNyako) [inline src/paddlefleet/fusions/fused_bias_swiglu.py]: 这里我暂时不能把它当成“等价重构”处理。

从当前 PR 的集成日志看，这个替换已经在多组 GLM4.5 场景里引入了可观测的精度漂移，并且直接触发了 `The precision has been changed and requires approvals.`：例如 H20 multi-card 的 `glm45_lora_multi_card_gt_loss` 最大绝对误差到 `0.01418428`、相对误差到 `1.52807667`，A100 的 `glm45_lora_multi_card_a100_gt_loss` 最大绝对误差到 `0.975`、相对误差到 `0.12721816`；即便是 PT 场景也有 `3e-05 ~ 6e-05` 的漂移。

建议先做两件事再合：
1. 明确确认 `paddle.nn.functional.swiglu` 在当前 Paddle 版本 / 目标硬件上的数值行为是否允许与原 `chunk + silu` 存在这些差异；
2. 补一个直接覆盖 `hidden_act=F.silu + gated_linear_unit=True + bias_activation_fusion=True` 的 forward/backward 回归单测，避免这条路径继续没有单测兜底。

如果这些精度变化是预期的，也请同步更新基线并走对应的 precision approval 流程。

**Review by ShigureNyako — CHANGES_REQUESTED** [review]
我先给 request changes。

这个 PR 的目标我理解为：把 `src/paddlefleet/fusions/fused_bias_swiglu.py` 里的手写 `chunk + silu` 前向实现替换成 Paddle 原生 `paddle.nn.functional.swiglu`，代码会更简洁。

但从当前证据看，它还不能直接按“无行为变化的重构”合入：
- 单元测试层面：现有单卡测试 `tests/single_card_tests/transformer/test_mlp.py` 的 `TestBiasFusedGatedMLP` 还没有显式走到 `hidden_act=F.silu + gated_linear_unit=True + bias_activation_fusion=True` 这条路径，对这次改动没有直接回归覆盖。
- CI 层面：`Unit test (single card)` 和 `Unit test (multi-card)` 已通过，但 3 个 integration job 都是 failure。日志里不是单纯环境报错，而是多个 GLM4.5 场景的 loss baseline 发生变化并触发了 `The precision has been changed and requires approvals.`。例如：
  - H20 multi-card `glm45_lora_multi_card_gt_loss`：max abs diff = `0.01418428`，max relative diff = `1.52807667`
  - A100 `glm45_lora_multi_card_a100_gt_loss`：max abs diff = `0.975`，max relative diff = `0.12721816`
  - H20/A100 PT 场景也分别出现了 `3.242e-05`、`6.294e-05` 量级的漂移

建议这一版至少补齐两件事：
1. 增加针对该融合路径的 forward/backward 回归测试，直接验证新旧实现的一致性；
2. 明确解释这些精度变化是否符合预期。若属于预期行为，请同步更新对应 baseline，并走 required 的 precision approval 流程后再合入。

File: src/paddlefleet/fusions/fused_bias_swiglu.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (huangjiyi) [reply src/paddlefleet/fusions/fused_bias_swiglu.py]: 补充更正一下：这版最终保留的是 fused 实现，不是回退到旧实现。

目前已补的覆盖是：
1. `TestBiasFusedGatedMLP` 显式设置 `hidden_act=F.silu`，保证不再默认走 `F.gelu`；
2. 新增 `TestBiasFusedSwiGLURegression`，直接覆盖 `hidden_act=F.silu + gated_linear_unit=True + bias_activation_fusion=True` 这条分支；
3. forward 对 `paddle.nn.functional.swiglu` 相对 `chunk + F.silu` 做容差回归，当前阈值是 `float32 <= 1e-6`、`float16 <= 2e-3`、`bfloat16 <= 7e-2`；backward 继续用 `fused_swiglu_bwd` 对解析 reference 做校验。

最新代码在 commit `009f4e5`。

Comment (huangjiyi) [reply src/paddlefleet/fusions/fused_bias_swiglu.py]: 补充更正一下：按这版 PR 的目标，最终代码侧保留 fused 路径，不回退到 `chunk + F.silu`。

我在本地 `.venv` / Paddle `3.3.0.dev20260123` 上做了 same-dtype probe，当前样例下 `paddle.nn.functional.swiglu` 相对 `chunk + F.silu` 的 forward `max abs diff` 为：
- `float32`: `2.384185791015625e-07`
- `float16`: `0.0009765625`
- `bfloat16`: `0.0625`

这版把这部分差异按 fused 实现的可接受数值波动处理，并新增了 `TestBiasFusedSwiGLURegression` 来固化容差边界：`float32 <= 1e-6`、`float16 <= 2e-3`、`bfloat16 <= 7e-2`。backward 仍然保留 `fused_swiglu_bwd` 对解析 reference 的校验。

如果后续 integration baseline 仍然提示 precision approval，再按 CI 流程更新 baseline / 走 approval；代码侧这版先不回退 fused 路径。最新 commit 是 `009f4e5`。

**Comment by huangjiyi** [comment]
已经在 https://github.com/PaddlePaddle/PaddleFleet/pull/714 中实现
